cmd/imagec/docker.go

```diff
@@ -183,45 +185,89 @@ func FetchImageBlob(options ImageCOptions, image *ImageWithMeta) error {
 	// Open the file so that we can use it as a io.Reader for sha256 calculation
 	in, err := os.Open(string(imageFileName))
 	if err != nil {
-		return err
+		return nil, err
 	}
 	defer in.Close()
 
-	h := sha256.New()
+	done := make(chan bool)
+	errs := make(chan error)
+	defer close(done)
+	defer close(errs)
+
+	r, w := io.Pipe()
+	t := io.TeeReader(in, w)
```

Comment:
This looks lopsided.  One of the go funcs uses `t` while the other uses `r`.  So, can you reduce by only using one of them (either Tee or Pipe)?